### ISSUE TITLE
Release failure in CI

### ISSUE THREAD

**Issue by Tieske** [lunarmodules/luasystem#13]
@alerque would you mind having a look at why the release jobs failed?

I tagged `v0.3.0` mostly worked, but seems the `0.2.1` version of the rockspec is failing to build. When I locally build that version it works for me?

It doesn't seem to find the source files, could it be that the monkey patching of the urls fails for some reason?

**Comment by alerque** [comment]
The issue is not `v0.2.1`, it is more specifically the `-0` rockrel for that release. The source details on that are for a tarball, our build system patches it to me a Git clone, but the rockspec also doesn't have the right tag field. Our monkey patch system is supposed to help us test build things from PRs and forks, but it is only meant for *our* rockspecs not ones that were released under other conditions.

I think the real question is why is it trying to build that rockspec at all since it wasn't touched in the commit and we shouldn't be trying to republish it...

**Comment by alerque** [comment]
Hmmm, it *was* touched since the previous push ... not in the final commit but in a commit leading up to it that hadn't landed in the remote repo yet before the v3 tag landed. Commit 0c8efdd moved the rockspec to a different directory. I think to avoid this problem that should have been pushed earlier and let to fail or succeed on it's own. It would still have failed, but at least it wouldn't have flunked the release process for the rockspec that actually matters.

I'll look into whether there is a way to untangle it now.

**Comment by alerque** [comment]
So somewhat unexpectedly but for our usual workflows probably correctly, the action that is detecting changed files looks at the last pushed commit for branch pushes but the last pushed tag for tags. Hence it is listing all rockspecs that changed between v0.2.1 and v0.3.0 which correctly finds that all the rockspecs got moved.

**Comment by alerque** [comment]
I fixed this and got it published with a somewhat un-intuitive fix:

I looked back at the previous actions run and instead of looking at the "deploy" runs for the tags (which should normally be the ones that successfully deploy) I opened the run for the branch. Normally when run this iteration of the deploy workflow will be ale to publish dev/scm rockspecs but not tagged ones because the branch push comes before the tag push and when run the tags corresponding to the rockspecs won't exist yet. That means they won't pass a test build.

In this case however re-running the workflow from the branch successfully deployed the new tagged rockspec version because the "changed files" delta only included the v0.3.0 rockspec AND the tag does now exist so it passed the test build.
